client/app/src/main/java/com/example/criengine/MyBooksActivity.java

```diff
@@ -64,6 +52,26 @@ protected void onCreate(Bundle savedInstanceState) {
         bookNameTextView = findViewById(R.id.bookListView);
         bookNameTextView.setAdapter(myBooksListAdapter);
 
+        // TODO: please pass this activity a profile object somehow, so we dont need to do this nightmare double call
+        dbw.getProfile(dbw.userId).addOnSuccessListener(
+                new OnSuccessListener<Profile>() {
+                    @Override
+                    public void onSuccess(Profile profile) {
+                        dbw.getOwnedBooks(profile).addOnSuccessListener(
+                                new OnSuccessListener<List<Book>>() {
+                                    @Override
+                                    public void onSuccess(List<Book> books) {
+                                        myBooks.addAll(books);
+                                        // TODO: This will definitely break filters, let's stop that somehow
+                                        displayBooks.addAll(myBooks);
+                                        myBooksListAdapter.notifyDataSetChanged();
```

Comment:
How does this break the filter?

Reply:
it adds every book in 'mybook' to display. if any filter is on when this loads, it ignores that filter. we should probably have a function for this at some point that looks like applyFilter(displayBooks, myBooks) that uses some sort of object prop to store the filter

Reply:
When this activity is loaded the filter array is going to be empty initially. Also since the information is retrieved in the OnCreate method, it will only ever be called once and if the user wants to see new books they have added in, they would need to refresh the page (which then wipes the filters).

It should not have any negative impact if my understanding is correct?